app/src/main/java/com/example/mystore/ui/home/filters/ExpandableList.kt

```diff
@@ -0,0 +1,132 @@
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import com.example.mystore.R
+import com.example.mystore.model.CategoryDetailsModel
+import com.example.mystore.ui.theme.Gray
+import com.example.mystore.ui.theme.Black
+import com.example.mystore.ui.theme.DarkPurple
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ExpandableList(
+    data: List<CategoryDetailsModel>,
+    selectedCategories: List<String>,
+    onFiltersChanged: (CategoryDetailsModel) -> Unit
+) {
+    var expanded by remember { mutableStateOf(false) }
+
+    val rotateState = animateFloatAsState(
+        targetValue = if (expanded) 180F else 0F,
+    )
+
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+    ) {
+
+        Card(
+            onClick = { expanded = !expanded },
+            shape = RectangleShape,
+            colors = CardDefaults.cardColors(
+                containerColor = Color.Transparent,
+            )
+        ) {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = CenterVertically
+            ) {
+                var selectedCategoriesString = stringResource(id = R.string.all_categories_filter)
+                val filteredCategories = data.filter { selectedCategories.contains(it.name) }
+                if (filteredCategories.isNotEmpty() && filteredCategories.count() != data.count()) {
+                    selectedCategoriesString = filteredCategories.joinToString { it.name }
+                }
+                Column {
+                    Text(
+                        text = stringResource(id = R.string.category),
+                        style = MaterialTheme.typography.bodyLarge,
+                        color = Black
+                    )
+                    Text(
+                        text = selectedCategoriesString,
+                        style = MaterialTheme.typography.labelMedium,
+                        color = Gray
+                    )
+                }
+                Icon(
+                    Icons.Outlined.ArrowDropDown, "",
+                    modifier = Modifier
+                        .rotate(rotateState.value)
+                        .size(dimensionResource(id = R.dimen.arrow_drop_down)),
+                    tint = Gray
+                )
+            }
+        }
+
+        AnimatedVisibility(
+            visible = expanded,
+        ) {
+            LazyColumn(
+                modifier = Modifier
+                    .fillMaxWidth()
+            ) {
+                items(data) { category ->
+                    Row(
+                        modifier = Modifier
+                           .padding(dimensionResource(id = R.dimen.padding_medium_20))
```

Comment:
`padding_medium_20` - try to disassociate numeric values from key naming conventions. This might become `22` some day and the key name will already be irrelevant.